pkg/mcp/source/source_test.go

```diff
@@ -639,3 +702,29 @@ func TestCalculateDelta(t *testing.T) {
 		})
 	}
 }
+
+type FakePerConnLimiter struct {
```

Comment:
Move this to /pkg/mcp/internal/test/rate_limiter.go?

Reply:
This was originally in the `internal/test` however I had to move it out since `Create()` returns RateLimit [interface](https://github.com/istio/istio/pull/11435/files#diff-86574868efda1b2f651475de96ae6c81R727) which is in internal package and this causes import cycle.

Reply:
Is there still an import cycle if `Create()` returns the concrete type `test.FakeRateLimiter` instead of the interface?

Reply:
hmm, I can't see how that would work. The signature for `Create()` is defined as follow:
```
type ConnectionRateLimit interface {
	Create() RateLimit
}
```
It actually returns an interface, I believe if `Create()` in `test.FakePerConnLimiter` returns a concrete type it would no longer adhere to  `ConnectionRateLimit` interface.